common/persistence/persistence-tests/queuePersistenceTest.go

```diff
@@ -66,24 +66,45 @@ func (s *QueuePersistenceSuite) TestDomainReplicationQueue() {
 
 	numMessages := 100
 	concurrentSenders := 10
-	messageChan := make(chan []byte)
+	messageChan := make(chan []byte, numMessages)
+	var publishErrors []error
+	var mu sync.Mutex
 
 	go func() {
 		for i := 0; i < numMessages; i++ {
-			messageChan <- []byte{1}
+			messageChan <- []byte{byte(i)}
 		}
 		close(messageChan)
 	}()
 
 	wg := sync.WaitGroup{}
 	wg.Add(concurrentSenders)
 
+	// Helper function for publishing with retry
+	publishWithRetry := func(message []byte) error {
+		var lastErr error
+		for i := 0; i < 3; i++ {
+			err := s.Publish(ctx, message)
+			if err == nil {
+				return nil
+			}
+			lastErr = err
+			time.Sleep(100 * time.Millisecond)
```

Comment:
let's reduce this to ~20ms just to not contribute too much to overall test duration

Reply:
Inside Publish function, we already have retries. You can increase retries there.